Support for TFS 2017u2, u3 and TFS2018rtm

jessehouwing opened the pull request:
Changes required to support:
https://github.com/tfsaggregator/tfsaggregator/pull/274

ARGH Only now found out I was on the wrong base branch. Rebasing and pushing....

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/tfsaggregator/tfsaggregator-core/4)
<!-- Reviewable:end -->

giuliov committed:
Fix NuGet Microsoft.TeamFoundationServer.ExtendedClient

giuliov committed:
Fix build: remove duplicate references from unit test project

giuliov committed:
Fix build: more tweaking of references

Aligned plugin with web

jessehouwing committed:
Converted to the 2017 project structure

jessehouwing committed:
Fix path to central version file

jessehouwing committed:
Remove duplicate Assembly attributes

jessehouwing committed:
Use new code to fetch project properties.

jessehouwing committed:
Bugfix for ProcessTemplateScope

jessehouwing committed:
Fixes path to icon

jessehouwing committed:
Added correc target framework version

jessehouwing committed:
Set target framework only when not supplied

jessehouwing committed:
Updating framework versions for unit test project

jessehouwing committed:
Move project output location to build props

dasMulli commented on Aggregator.Core.Plugin/Aggregator.Core.Plugin.csproj:
```diff
@@ -0,0 +1,201 @@
+    <AssemblyName>TFSAggregator2.Core</AssemblyName>
+    <TargetFramework Condition="'$(TargetFramework)' == ''">net461</TargetFramework>
+    <FileAlignment>512</FileAlignment>
+    <Configurations>Debug-2013;Release-2013;Debug-2015;Release-2015;Debug-2015.1;Release-2015.1;Debug-2015.2;Release-2015.2;Debug-2017;Release-2017;Debug-2017.2;Release-2017.2;Debug-2018;Release-2018</Configurations>
```
Ideally, there shouldn't be any `Debug-XYZ` / `Release-XYZ` configurations left if all SDK projects inherit a default `TfsVersion` parameter from a `Directory.Build.props` file (auto-imported). The properties could then be conditioned on the `TfsVersion` parameter instead of the configuration (or a set in the Directory.Build.props file).

This would allow adding/removing tfs versions without fiddling with solution and projects configurations.

jessehouwing replied:
I tried that, But Visual Studio doesn't pick that up and it breaks the compiler directives in the IDE. But it is a nice next step. For now I'm glad it actually builds. I also tried auto-importing TargetFramework, but that triggers an Upgrade Wizard in VS.

jessehouwing replied:
I tried provisioning the <configurations> property using a filter expression using the TfsVersions, but the IDE doesn't like that either, it auto-generates all the project configs on the fly.

dasMulli replied:
Are the configurations still in the solution? I'd suggest starting out with a completely new solution file. Or leave it at the status quo.

jessehouwing replied:
I think for now I'll leave it. I'm working on removing as much as possible from the configurations, to make it as easy as I can to maintain them. copy-pasting 2 sections into 6 files every 3 months is doable.

jessehouwing commented:
How else would visual studio know how to switch between tfs versions when
editing?

On 21 Jan 2018 12:39, "Martin Andreas Ullrich" <notifications@github.com>
wrote:

> *@dasMulli* commented on this pull request.
> ------------------------------
>
> In Aggregator.Core.Plugin/Aggregator.Core.Plugin.csproj
> <https://github.com/tfsaggregator/tfsaggregator-core/pull/4#discussion_r162809007>
> :
>
> > @@ -0,0 +1,201 @@
> +﻿<?xml version="1.0" encoding="utf-8"?>
> +<Project Sdk="Microsoft.NET.Sdk">
> +  <PropertyGroup>
> +    <Configuration Condition=" '$(Configuration)' == '' ">Debug-2018</Configuration>
> +    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
> +    <ProjectGuid>{FEA76835-61B3-4FC7-9D6F-50D39A7A8258}</ProjectGuid>
> +    <OutputType>Library</OutputType>
> +    <AppDesignerFolder>Properties</AppDesignerFolder>
> +    <RootNamespace>Aggregator.Core</RootNamespace>
> +    <AssemblyName>TFSAggregator2.Core</AssemblyName>
> +    <TargetFramework Condition="'$(TargetFramework)' == ''">net461</TargetFramework>
> +    <FileAlignment>512</FileAlignment>
> +    <Configurations>Debug-2013;Release-2013;Debug-2015;Release-2015;Debug-2015.1;Release-2015.1;Debug-2015.2;Release-2015.2;Debug-2017;Release-2017;Debug-2017.2;Release-2017.2;Debug-2018;Release-2018</Configurations>
>
> Are the configurations still in the solution? I'd suggest starting out
> with a completely new solution file. Or leave it at the status quo.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/tfsaggregator/tfsaggregator-core/pull/4#discussion_r162809007>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AD-uSwxPcfbr23gUKZEs0KDn2ijNklxZks5tMyIMgaJpZM4RlxGo>
> .
>

dasMulli commented:
> How else would visual studio know how to switch between tfs versions when
editing?

I would have suggested requiring the user to edit the current "development target" in Directory.Build.props (include as solution item) or move it to a "devsettings.props". If that doesn't work properly or is considered too awkward, then just leave it at configurations.

jessehouwing committed:
Cleaning up project files

Removed outout path config from poject file. Now centrally controlled.
Removed dependency on AssemblyVersion.cs, this is now centrally
controlled.
Cleaned up and re-ordered project configurations

jessehouwing committed:
Removing packages.config from project file
